tests/Acceptance/DependencyInjection/ConfigurationTest.php

```diff
@@ -46,13 +50,43 @@ public function invalid(array $config, ?string $expectedMessage = null): void
 
     /**
      * @test
+     *
      * @dataProvider provideValidConfig
      */
     public function valid(array $config): void
     {
         $this->assertConfigurationIsValid($config);
     }
 
+    /**
+     * @test
+     */
+    public function addConfig()
+    {
+        $configProvider = ExampleConfiguration::provideDefaultConfig();
+
+        $config = new ConfigDTO('new-public', 'new-private');
+        $configProvider->addConfig('new_config', $config);
+
+        $this->assertEquals('public-key', $configProvider->getPublicKey());
+
+        $configProvider->setConfig('new_config');
+
+        $this->assertEquals('new-public', $configProvider->getPublicKey());
+    }
+
+    /**
+     * @test
+     */
+    public function addConfigWithColidingName()
+    {
+        $this->expectExceptionMessage('Configuration with key "config_name" already exists.');
+        $configProvider = ExampleConfiguration::provideDefaultConfig();
+
+        $config = new ConfigDTO('public', 'private');
+        $configProvider->addConfig('config_name', $config);
+    }
```

Comment:
ConfigurationTest is a test for dependencyInjection configuration.
Maybe you can move this tests for another class ex. `\Answear\LuigisBoxBundle\Tests\Unit\DTO\ConfigDTOTest`?